src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/NullableConverter.cs

```diff
@@ -27,13 +30,16 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
                 return true;
             }
 
-            state.Current.JsonPropertyInfo = state.Current.JsonTypeInfo.ElementTypeInfo!.PropertyInfoForTypeInfo;
-            if (_elementConverter.TryRead(ref reader, typeof(T), options, ref state, out T element))
+            JsonTypeInfo previousTypeInfo = state.Current.JsonTypeInfo;
```

Comment:
Why do we no longer need to set `state.Current.JsonPropertyInfo`?

Reply:
when we transition from `Converter<Nullable<T>>` to `Converter<T>` property stays the same (it's compatible to use Set etc)